### PR TITLE
[Gradient checkpoining] Correct disabling `find_unused_parameters` in Trainer when gradient checkpointing is enabled

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Following https://github.com/huggingface/transformers/pull/13657 this PR makes sure that the Trainer uses the new gradient_checkpointing logic to disable `find_unused_parameters` argument in DDP. 

@sgugger - don't think many people have switched from `self.config._gradient_checkpointing` to the new API yet, but for those that have previously `find_unused_parameters` would not have been set to `False` which then leads to some hard to debug problems like: https://discuss.pytorch.org/t/finding-the-cause-of-runtimeerror-expected-to-mark-a-variable-ready-only-once/124428/5

Not sure if this worth a patch or not

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing! The check in the `Trainer` was changed to match the intermediate iteration in the PR for gradient checkpointing and I forgot to adapt it to the last one.
